Source/AwsCommonRuntimeKit/sdkutils/CRTAWSProfile.swift

```diff
@@ -3,11 +3,14 @@
 
 import AwsCSdkUtils
 
-public struct CRTAWSProfile {
+public class CRTAWSProfile {
     let rawValue: OpaquePointer
+    // Keep a reference of collection to keep it alive
+    let collection: CRTAWSProfileCollection
```

Comment:
oh, sneaky. nice simple fix

This is how we USED TO do the bindings, before we added ref-counting to everything. If a few different objects all had their lifetimes tied together, then the binding for one object would keep pointers to the bindings for all the other objects too, to keep them alive.

It starts falling apart when the web of entanglements gets too complex, or things start having async shutdown procedures, but it works here 👍